src/shims/mod.rs

```diff
@@ -65,4 +46,39 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         // Otherwise, load the MIR.
         Ok(Some(this.load_mir(instance.def, None)?))
     }
+
+    fn align_offset(
+        &mut self,
+        ptr_op: OpTy<'tcx, Tag>,
+        align_op: OpTy<'tcx, Tag>,
+        layout: ty::layout::TyLayout<'tcx>,
+    ) -> InterpResult<'tcx, u128> {
```

Comment:
Maybe return `Option<u128>` to leave the handling of "we do not (want to) support this case" up to the caller?

Reply:
So basically we return `None` instead of the `u128::max_value()`, right?

Reply:
That's what I was thinking. But only if it simplifies/clarifies anything -- not sure if it actually is a good ideal

Reply:
I think it would allow to do the truncate part clearer